src/test/java/org/mule/extension/aggregator/AsyncTasksAfterRestartTestCase.java

```diff
@@ -182,6 +184,87 @@ public void sizeBasedAggregatorAsyncTasksAfterRedeploy() throws Exception {
     assertAsyncAggregation(sourceCallback, sourceCallbackCtx);
   }
 
+  @Test
+  @Description("Avoid race condition in object store after redeploy in sizeBasedAggregator")
+  public void sizeBasedAggregatorDoesNotPushToObjectStoreWhenNoChanges() throws Exception {
+    final Map<String, Object> params = new HashMap<>();
+    SimpleMemoryObjectStore mockedObjectStore = spy(SimpleMemoryObjectStore.class);
+    params.put("objectStore", mockedObjectStore);
+    params.put("name", "aggregator");
+    params.put("maxSize", 2);
+
+    final SizeBasedAggregatorOperationsExecutor aggregatorExecutor = new SizeBasedAggregatorOperationsExecutor(params);
+
+    initialiseIfNeeded(aggregatorExecutor, muleContext);
+    startIfNeeded(aggregatorExecutor);
+
+    final AggregatorListener aggregatorListener = mock(AggregatorListener.class);
+    when(aggregatorListener.getCallback()).thenReturn(sourceCallback);
+    when(aggregatorListener.shouldIncludeTimedOutGroups()).thenReturn(true);
+    aggregatorManager.registerListener("aggregator", aggregatorListener);
+
+    verify(mockedObjectStore, never()).store(any(), any());
+
+    final Map<String, Object> operationParams = new HashMap<>();
+    operationParams.put("content", new TypedValue<>("testPayload", DataType.STRING));
+    operationParams.put("timeout", RECEIVE_TIMEOUT);
+    operationParams.put("timeoutUnit", MILLISECONDS);
+    addItemToAggregation(aggregatorExecutor, operationParams);
+
+    // Wait until just before the aggregation times out to do a restart...
+    Thread.sleep(3000);
+
```

Comment:
If you need to wait for a condition to be met, perhaps use polling probers.

Reply:
These Thread.sleep() were not necessary.